pb-cli.js

```diff
@@ -1,791 +1,1668 @@
-    }
-    else
-	console.log('usage: atalk <text...>');
-}
-
-// Chat mode
-else if (program.args[0] === 'chat') {
-    console.log('Entering chat with ' + conf_botname());
-    console.log('Press Control-C at any time to exit');
-    var rl = readline.createInterface({
-        input: process.stdin,
-        output: process.stdout
-    });
-    rl.setPrompt('user> ');
-    rl.prompt();
-    rl.on('line', function(cmd) {
-        var param = {input: cmd};
-        if (nconf.get('client_name'))
-            param.client_name = nconf.get('client_name');
-        if (nconf.get('sessionid'))
-            param.sessionid = nconf.get('sessionid');
-        request.post({url: talkUri(), form: composeParams(param)}, updatedChatResp(cmd));
-    });
-}
-
-// Achat mode
-else if (program.args[0] === 'achat') {
-    console.log('Entering achat with ' + conf_botname());
-    console.log('Press Control-C at any time to exit');
-    var rl = readline.createInterface({
-        input: process.stdin,
-        output: process.stdout
-    });
-    rl.setPrompt('user> ');
-    rl.prompt();
-    rl.on('line', function(cmd) {
-        var param = {input: cmd};
-        if (nconf.get('client_name'))
-            param.client_name = nconf.get('client_name');
-        if (nconf.get('sessionid'))
-            param.sessionid = nconf.get('sessionid');
-        request.post({url: atalkUri(), form: composeParams(param)}, updatedChatResp(cmd));
-    });
-}
-
-else if (program.args[0] === undefined)
-    console.log('usage: <command> [--options...]');
-else {
-    console.log('invalid command: ' + program.args[0]);
-    process.exit(1);
-}
+
+#!/usr/bin/env node
+
+'use strict';
+
+var fs = require('fs');
+var readline = require('readline');
+var glob = require('glob');
+var nconf = require('nconf');
+var path = require('path');
+var program = require('commander');
+var prompt = require('prompt');
+var qs = require('querystring');
+var request = require('request');
+var unzip = require('unzip');
+var url = require('url');
+var Filequeue = require('filequeue');
+
+var fq = new Filequeue(500);
+
+var config = 'chatbot.json';
+var errors = require('./errors.js')
+
+var sep = function (str) {
+    return str ? '/' + str : '';
+}
+
+var composeUri = function (mode, botname, kind, filename) {
+  console.log("compose uri called");
+    var uri = {
+	protocol: nconf.get('protocol'),
+	hostname: nconf.get('hostname'),
+	port: nconf.get('port')
+    };
+    var path = '';
+    path += sep(nconf.get('prefix'));
+    path += sep(mode);
+    path += sep(conf_app_id());
+    path += sep(botname);
+    path += sep(kind);
+    path += sep(filename);
+    uri.pathname = path;
+    return uri;
+}
+
+var prioritizeMode = function(){
+  var ukey = nconf.get('user_key');
+  var akey = nconf.get('app_id');
+  var bname = nconf.get('botkey');
+  if(ukey && akey && bname){
+    return 1;
+  }else{
+    if(bname){
+      return 1;
+    }else{
+      return 0;
+    }

```

Comment:
This is a completely correct way of writing this function, but as a nitpicky style tip it would be more common in Javascript to return `true` and `false` instead of `1` and `0`. 

This goes along with my comment about function naming (see below) - `usingBotkey() -> true` would make it very obvious at a glance what the function is doing.